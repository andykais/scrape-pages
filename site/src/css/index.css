:root {
  --primary: #78909c;
  --primary-dark: #4b636e;
  --secondary: #bcaaa4;
  --secondary-light: white;
  --flow-error: #c40;
  --font-size: 16px;
}

html,
body {
  font-size: var(--font-size);
  padding: 0;
  margin: 0;
  font-family: sans-serif;
  height: 100%;
}
body {
  display: flex;
  flex-direction: column;
}
h1,
h2,
h3,
h4,
h5 {
  margin: 0;
}
a {
  text-decoration: none;
  position: relative;
  color: white;
}
i {
  font-size: 26px;
  line-height: 20px;
}
button i {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

#container {
  display: flex;
  flex-grow: 1;
}

@import './navbar.css';
@import './user-input.css';
@import './flow-diagram.css';
