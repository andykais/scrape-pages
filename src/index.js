import PageScraper from './classes'
import sourceMapSupport from 'source-map-support'
sourceMapSupport.install()

// TODO move this into a try catch around `run`
// I shouldnt be blocking unhandledRejections here
process.on('unhandledRejection', error => {
  console.error(error)
  process.exit(1)
})

export default PageScraper
