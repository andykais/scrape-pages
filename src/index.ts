export { scrape } from './scraper'

export * from './util/errors'

export * from './util'
