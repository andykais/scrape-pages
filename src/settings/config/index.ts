export { normalizeConfig } from './normalize'

export { flattenConfig } from './flatten'

export { assertConfigType } from './runtime/settings/config/assert'
