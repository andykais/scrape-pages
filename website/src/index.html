<!DOCTYPE html>
<html>
  <head>
    <meta name="viewport" content="initial-scale=1.0, width=device-width">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.2.0/css/all.css" integrity="sha384-hWVjflwFxL6sNzntih27bfxkr27PmbbK/iSvJ+a4+0owXq79v+lsFkW54bOGbiDQ" crossorigin="anonymous">
  </head>
  <body>
    <div id="navbar">
      <div id="title-container">
        <h2>scrape-pages</h2>
        <h5>config editor</h5>
      </div>
      <div class="icons">
        <span id="share">
          <i class="fas fa-share-alt"></i>
        </span>
        <a href="https://github.com/andykais/scrape-pages">
          <i class="fab fa-github"></i>
        </a>
      </div>
    </div>
    <div id="container">
      <div id="user-input">
        <textarea id="config-input" name="config-input" cols="30" rows="10"></textarea>
        <button id="normalize-button">
          <span>
            Normalize Config
          </span>
        </button>
      </div>
      <div id='hide-button-container'>
        <button id="hide-button">
          <i class="fas fa-caret-left"></i>
        </button>
      </div>
      <div id="config-dependent-content">
        <div id="tree-container"></div>
        <div id="flow-error-block"></div>
      </div>
    </div>
  </body>
</html>
