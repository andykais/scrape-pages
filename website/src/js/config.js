const config = {
  input: [],
  scrape: {
    name: "mainpage",
    download: {
      increment: 0,
      initialIndex: 0,
      headerTemplates: {},
      cookieTemplates: {},
      method: "GET",
      urlTemplate: "mainpage"
    },
    parse: { expect: "html", selector: "batchID" },
    scrapeEach: [
      {
        name: "grid",
        download: {
          increment: 1,
          initialIndex: 0,
          headerTemplates: {},
          cookieTemplates: {},
          method: "GET",
          urlTemplate: "batchPage{batchID}"
        },
        scrapeNext: {
          name: 'next-batchid',
          parse: {
            selector: "batchID",
            expect: "html"
          }
        },
        scrapeEach: [
          {
            name: "tag",
            parse: { expect: "html", selector: "span.tag" },
            scrapeEach: []
          },
          {
            name: "parse-image-page",
            parse: { expect: "html", selector: "li a.images" },
            scrapeEach: [
              {
                name: "image-page",
                download: {
                  increment: 0,
                  initialIndex: 0,
                  headerTemplates: {},
                  cookieTemplates: {},
                  method: "GET",
                  urlTemplate: "imagepage{imageID}"
                },
                parse: { expect: "html", selector: "img" },
                scrapeEach: [
                  {
                    name: "image",
                    download: {
                      increment: 0,
                      initialIndex: 0,
                      headerTemplates: {},
                      cookieTemplates: {},
                      method: "GET",
                      urlTemplate: "{src}"
                    },
                    scrapeEach: []
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
};
