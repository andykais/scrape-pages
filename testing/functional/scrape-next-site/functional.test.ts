import os from 'os'
import path from 'path'

import { expect } from 'chai'
import _ from 'lodash/fp'

import '../../use-chai-plugins'
import { nockMockFolder } from '../../nock-folder-mock'
import { config } from './config'
import expectedQueryResult from './resources/expected-query-result.json'
import { scrape } from '../../../src'

describe('scrape next site', () => {
  describe('with instant scraper', function() {
    let scraperQueryForFunction: any
    before(done => {
      ;(async () => {
        await nockMockFolder(
          `${__dirname}/resources/mock-endpoints`,
          'http://scrape-next-site.com'
        )

        const options = {
          folder: path.resolve(os.tmpdir(), this.fullTitle()),
          cleanFolder: true,
          optionsEach: {
            image: {
              read: false,
              write: true
            }
          }
        }
        const { on, query } = await scrape(config, options)
        scraperQueryForFunction = query
        on('done', done)
      })()
    })

    it('should group each image into a separate slot, in order', () => {
      const result = scraperQueryForFunction({
        scrapers: ['image'],
        groupBy: 'image'
      })
      expect(result)
        .excludingEvery('filename')
        .to.be.deep.equal(expectedQueryResult.map(_.omit('tag')))
    })
    it('should group tags and images together that were found on the same page', () => {
      const result = scraperQueryForFunction({
        scrapers: ['image', 'tag'],
        groupBy: 'image-page'
      })
      expect(result)
        .excludingEvery('filename')
        .to.be.deep.equal(expectedQueryResult)
    })
  })

  describe('with psuedo-random delayed scraper', function() {
    let scraperQueryForFunction: any
    before(done => {
      ;(async () => {
        await nockMockFolder(
          `${__dirname}/resources/mock-endpoints`,
          'http://scrape-next-site.com',
          { randomSeed: 2 }
        )

        const options = {
          folder: path.resolve(os.tmpdir(), this.fullTitle()),
          cleanFolder: true,
          optionsEach: {
            image: {
              read: false,
              write: true
            }
          }
        }
        const { on, query } = await scrape(config, options)
        scraperQueryForFunction = query
        on('done', done)
      })()
    })

    it('should keep images and tags together, in order', () => {
      const result = scraperQueryForFunction({
        scrapers: ['image', 'tag'],
        groupBy: 'image-page'
      })
      expect(result)
        .excludingEvery('filename')
        .to.be.deep.equal(expectedQueryResult)
    })
  })
})

/**
 * I need a way to say, use the cached response from certain scrapers if hit, and always download again for
 * others
 * - useCachedResponse? default to true?
 */
