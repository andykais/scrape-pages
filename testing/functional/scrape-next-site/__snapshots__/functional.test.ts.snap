// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should group each image into a separate slot, in order 1`] = `
Array [
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`should group each image into a separate slot, in order 2`] = `
Array [
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`should group each image into a separate slot, in order 3`] = `
Array [
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`should group tags and images together that were found on the same page 1`] = `
Array [
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "three",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`should group tags and images together that were found on the same page 2`] = `
Array [
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "three",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`should group tags and images together that were found on the same page 3`] = `
Array [
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "three",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`testing/functional/scrape-next-site/functional.test.ts with flattened config scraper should group each image into a separate slot, in order 1`] = `
Array [
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`testing/functional/scrape-next-site/functional.test.ts with flattened config scraper should group tags and images together that were found on the same page 1`] = `
Array [
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "three",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`testing/functional/scrape-next-site/functional.test.ts with normal scraper should group each image into a separate slot, in order 1`] = `
Array [
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`testing/functional/scrape-next-site/functional.test.ts with normal scraper should group tags and images together that were found on the same page 1`] = `
Array [
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "three",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`testing/functional/scrape-next-site/functional.test.ts with psuedo delayed scraper should group each image into a separate slot, in order 1`] = `
Array [
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;

exports[`testing/functional/scrape-next-site/functional.test.ts with psuedo delayed scraper should group tags and images together that were found on the same page 1`] = `
Array [
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/the.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "one",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "two",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/quick.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/brown.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
  Array [
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "three",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "four",
      "scraper": "tag",
    },
    Object {
      "byteLength": null,
      "complete": 1,
      "downloadData": null,
      "filename": false,
      "id": true,
      "parsedValue": "five",
      "scraper": "tag",
    },
    Object {
      "byteLength": "17.0",
      "complete": 1,
      "downloadData": "[\\"http://scrape-next-site.com/image/fox.jpg\\",{\\"headers\\":{},\\"method\\":\\"GET\\"}]",
      "filename": true,
      "id": true,
      "parsedValue": "",
      "scraper": "image",
    },
  ],
]
`;
